### PR TITLE
WIP: DEMO: make dangling unwanted MachineConfig

### PR DESCRIPTION
**Pull request by ffromani**
is this better than the current approach?

### CODE REVIEW

**Comment by ffromani** [comment]
this can be clever, but I think it's better to delete explicitely, to really emphasize that we want the extra MachineConfig to be gone.

From user perspective the overall behavior should be the same.

File: internal/dangling/dangling.go [hunk 1]
```diff
@@ -117,6 +119,21 @@ func DeleteUnusedMachineConfigs(cli client.Client, ctx context.Context, instance
 	return errors
 }
 
+func keepMachineConfig(expected bool, mc machineconfigv1.MachineConfig, filterObjects ...KeepDanglingObjectFunc) bool {
+	if len(filterObjects) == 0 {
+		return expected
+	}
+	for _, filterObject := range filterObjects {
+		keep, reason := filterObject(expected, &mc)
+		if keep {
```
Comment (Tal-or) [inline internal/dangling/dangling.go:128]: I don't see any good reason for keeping a dangling object.
Moreover we should expose a way for the user to specify whether he wants that or not.

**Review by Tal-or — COMMENTED** [review]
Could you please explain the rationale here? I'm probably missing something

**Comment by ffromani** [comment]
this was a demo to explore if it's better to use the dangling objects handling flow or the current explicit delete flow.
While the former is feasible, the current solution seems better as it is more explicit.
So no longer need for this PR.

**Pull request closed without merging**